### PR TITLE
Implements erc165

### LINKED ISSUE #758: We must implement the ERC165 interface as part of achieving ERC721 compatibility.
**Is your feature request related to a problem? Please describe.**
We want unlock-protocols "keys" to be compliant rerc721 NFT's. This is one of the required steps. It allows our contract to be queried to determine if we're erc721 compliant. Obviously, there's no point in doing this in isolation, but it's one step that needs to be done. 

**Describe the solution you'd like**
We need to inherit the erc165 interface and implement its single function.
```
interface ERC165 {
    /// @notice Query if a contract implements an interface
    /// @param interfaceID The interface identifier, as specified in ERC-165
    /// @dev Interface identification is specified in ERC-165. This function
    ///  uses less than 30,000 gas.
    /// @return `true` if the contract implements `interfaceID` and
    ///  `interfaceID` is not 0xffffffff, `false` otherwise
    function supportsInterface(bytes4 interfaceID) external view returns (bool);
}
```

**Additional context**
Depends on #148, #748, #759
https://github.com/ethereum/EIPs/blob/master/EIPS/eip-721.md

### PR DESCRIPTION
**Pull request by nfurfaro**
Inherited the oz eth contract in PublicLock.sol.
Added tests to make sure erc165 and erc721 are implemented.

Fixes #758
## Important!
Prior to merging (assuming we want to proceed), I would like to test that inheriting a new interface and contract will not present any problems with upgradeability.
## Proposed Changes

- Implement erc165 spec (required for erc721 compatibility)
- Add 2 new tests

### CODE REVIEW

**Comment by nfurfaro** [comment]
Note: I had problems with my branch, so this is a fresh PR. sorry about that.

File: smart-contracts/build/contracts/Unlock.json [hunk 1]
```diff
@@ -4248,8 +4248,14 @@
       "events": {},
       "address": "0xd8c88be5e8eb88e38e6ff5ce186d764676012b0b",
```
Comment (julien51) [inline smart-contracts/build/contracts/Unlock.json]: Ok, this is preserved so I am tempted to say we're all good here ;p

**Comment by nfurfaro** [comment]
@julien51 well well. finally.

File: package-lock.json [hunk 2]
```diff
@@ -1092,8 +1092,7 @@
             "bindings": {
               "version": "1.3.0",
               "resolved": "https://registry.npmjs.org/bindings/-/bindings-1.3.0.tgz",
-              "integrity": "sha512-DpLh5EzMR2kzvX1KIlVC0VkC3iZtHKTgdtZ0a3pglBZdaQFjt5S9g9xd1lE+YvXyfd6mtCeRnrUfOLYiTMlNSw==",
-              "dev": true
```
Comment (nfurfaro) [inline package-lock.json]: Any Idea why this would have changed?

**Comment by julien51** [comment]
Yay!

File: package-lock.json (hunk 2 above)
Comment (nfurfaro) [reply package-lock.json]: cc @julien51

Comment (julien51) [reply package-lock.json]: This is something I have been trying to understand for weeks. npm is weird. this has no impact.

Comment (nfurfaro) [reply package-lock.json]: Ok, thx.

**Pull request merged**